ls1intum/Artemis
# `Exam mode`: Reject exam exercises with date attributes set

### Checklist

### Motivation and Context
Fix #3916.

### Description
- [x] Add many tests for all different REST endpoints that expect a bad request response for exercises that have dates
- [x] Make all the tests pass

### Steps for Testing


Instructor:
1. Create an exam with exercise groups
2. Create a couple of new exercises (all types) and make sure it works correctly
3. Import a couple of exercises (from other exams and other courses, all exercise types) and make sure it works without issues.
4. Import course exercises in which all date attributes (release, due, assessment due) are set. When importing them into an exam, the date attributes should be automatically be removed

### Review Progress





- Code Review
- Manual Tests

### Test Coverage




### Screenshots
